### PR TITLE
Redesign Slider and add a11y support

### PR DESCRIPTION
**Pull request by nathanbrud**
Initial comments in this PR - https://github.com/vimeo/iris/pull/280

https://vimean.atlassian.net/browse/DSYS-4
https://vimean.atlassian.net/browse/DSYS-5
**Storybook:** https://vimeo.github.io/iris/sb/DSYS-4-DSYS-5-slider-updates/?path=/story/components-slider--controls

## What this PR does 
* Change the design on Slider component based on the new approved design.
* Added values in onDragEnd callback
* Added support for a11y - arrow keys and home/end keys support.
* Fixed a bug where min / max can overlap each other.
* Added unit tests

## Screenshots & Recordings 
https://github.com/vimeo/iris/assets/9106375/1bddfc1b-5832-4f30-8061-d5c8857df185

## How it does that 
* Move input out from the `Handle` component into the main `Slider` component.
* Remove unused styling.
* Added 2 more focus types for inputs so it won't collide with keydown events.
* Better manage focus so it can react to key events.

## Testing 
image

### CODE REVIEW

**Review by juliewongbandue — CHANGES_REQUESTED** [review]
Hey @nathanbrud! I've taken a look at this in both Chrome (114.0.5735.133) and Firefox (114.0.2 ) and I'm unable to tab to the handles for both the single and range sliders. They must be clicked onto before being able to use the arrow keys to increment ... are you able to tab to the handles on your end?

It looks like the visual focus state on either handle was removed here as well: 

https://github.com/vimeo/iris/assets/22207955/c1df9274-3773-4573-b53b-76110aa65950

ref: https://www.w3.org/WAI/ARIA/apg/patterns/slider/

**Comment by nathanbrud** [comment]
Thanks @juliewongbandue 
I fixed the issues and moved some logic back to `Handle` component.
Focus should be visible on keyboard events only, as per the requested design. (same as we did with Button)
https://github.com/vimeo/iris/assets/9106375/e7dadf5f-0270-4d86-ae0a-e54d707bd366

File: src/components/inputs/Slider/Slider.tsx [hunk 1]
```diff
@@ -35,50 +54,101 @@ export function Slider({
...
+
+      const action =
+        focused === 'startHandle' ? setStartValue : setEndValue;
+      const currentValue =
+        focused === 'startHandle' ? values[0] : values[1];
+
+      switch (event.keyCode) {
+        case arrowDown:
+        case arrowLeft:
+          action(currentValue - 1);
```
Comment (juliewongbandue) [inline src/components/inputs/Slider/Slider.tsx:130]: There isn't a case in which we'd need to increment these by a different value aside from `1`, is there?

**Review by juliewongbandue — DISMISSED** [review]
Thanks for applying those a11y changes! Just one non-blocking question.

File: src/components/inputs/Slider/Slider.tsx (hunk 1 above)
Comment (nathanbrud) [reply src/components/inputs/Slider/Slider.tsx:130]: Based on the specs [here](https://www.w3.org/WAI/ARIA/apg/patterns/slider/), it's either 1 or the max / min value.
Both flows are supported in this PR :)

**Comment by nathanbrud** [comment]
Hi @juliewongbandue, after testing in real life, I found an issue with margins and a double focus, fixed them both.
Thanks!

**Comment by juliewongbandue** [comment]
As noted offline: this just needs a rebase to publish a prerelease since we have bumped a minor version.

**Review by juliewongbandue — APPROVED** [review]

**Pull request merged**